search-spec/openapi.yml

```diff
@@ -0,0 +1,229 @@
+        priority:
+          $ref: '#/components/schemas/Priority'
+        price:
+          $ref: '#/components/schemas/Price'
+        lifecycle:
+          type: array
+          example: [ "FEASIBILITY", "PRICE", "DELIVERY" ]
+          items:
+            type: string
+    Temporal:
+      type: object
+      required:
+        - interval
+      properties:
+        interval:
+          type: array
+          minItems: 1
+          items:
+            type: array
+            minItems: 2
+            maxItems: 2
+            items:
+              type: string
+              nullable: true
+              format: date-time
+              pattern: (\\+00:00|Z)$
+    Spatial:
+      type: object
+      required:
+        - geometry
+      properties:
+        geometry:
+          $ref: https://geojson.org/schema/Geometry.json
+    TaskingProperties: # this is everything that's in slotcollection and affects number of slots
+      type: object
+      description: "Satellite properties that affect number of slots. 
+      Every property of type array creates slotcollection for every combination with other fields. Every property of type range doesn't, i.e. 
+      all existing slot collection need to be within range"
+      properties:
+        provider:
+          type: array
+          items:
+            type: string
+          example: ["Airbus", "Capella"]
+        satelliteType:
+          type: array
+          items:
+            type: string
+          enum: [ "SAR", "Hyperspectral", "Optical", "Aerial" ]
+          example: [ "Optical" ]
+        sensor:
+          type: array
+          items:
+            type: string
+          example: [ "PHR", "SPOT" ]
+        imageMode:
+          type: array
+          items:
+            type: string
+            example: StripMap
+          example: [ "StripMap", "SpotLight" ]
+        incidenceAngle:
+          type: array
+          minItems: 1
+          example: [0, 5]
+          items:
+            type: array
+            minItems: 2
+            maxItems: 2
+            items:
+              type: integer
+              nullable: true
+        cloudCoverage:
+          type: array
+          minItems: 1
+          example: [0, 100]
+          items:
+            type: array
+            minItems: 2
+            maxItems: 2
+            items:
+              type: integer
+              nullable: true
+    ProductConfiguration: # this is everything thats is in slotcollection
+      type: object
+      description: "Values which do not effect the slots availibility"
+      properties:
+        processingLevel:
+          type: string
+        imageFormat:
+          type: string
+        resolution:
+          type: string
+          example: uint16
+    Price:
+      type: array
+      minItems: 1
+      example: [ 0, 1000 ]
+      items:
+        type: array
+        minItems: 2
+        maxItems: 2
+        items:
+          type: integer
```

Comment:
Should individual price consist of a value and a unit? To enable prices in different currencies (and potentially even synthetic units like [UP42 credits](https://docs.up42.com/getting-started/credits/purchase#overview)?).